### ISSUE TITLE
Upgrading from v4.0.6 to v4.0.7 failed

### ISSUE THREAD

**Issue by ernsyn** [jhipster/generator-jhipster#5366]
##### **Overview of the issue**

I tried to upgrade my JHipster app v4.0.6 to v4.0.7 but it failed when running what I presume to be "gulp install". 

##### **Reproduce the error**

To reproduce, simply run "yo jhipster:upgrade" on an existing app that was created using JHipster v4.0.6. I'm not sure whether other versions are affected or not. The screenshot below shows the error.

![jhipster-upgrade-output](https://cloud.githubusercontent.com/assets/6326517/23588175/8e2052a2-01f4-11e7-918d-1b5c71254602.png)

##### **Suggest a Fix**

Did some troubleshooting and found out that the error was due to 2 files having some upgrade remnants left over. Both files had "<<<<<<< HEAD", ">>>>>>> jhipster_upgrade" causing the build to fail. The 2 files are: 

1. gulpfile.js
![gulpfile-snippet](https://cloud.githubusercontent.com/assets/6326517/23588136/f55b141c-01f3-11e7-9c84-f245a9920e15.png)

2. pom.xml
![pom-xml-snippet](https://cloud.githubusercontent.com/assets/6326517/23588137/f55cc6a4-01f3-11e7-8f8b-032091a6510f.png)

Manually cleaning up both files fixes the build and I tested the application after that and they seem to run fine. However, I'm not sure what other steps did the upgrade missed due to this error. (When this error happen, I'm back to the master branch with lots of uncommitted files)

##### **JHipster Version(s)**

Currently using JHIpster v4.0.6.

##### **Browsers and Operating System**

macOS Sierra

**Comment by gmarziou** [comment]
Looks like merging failed, what does print `git --version`?

**Comment by jdubois** [comment]
Thank you for your bug report -> that's why this sub-generator is still marked as "not working" on the website for JHipster 4.x.
So we know it's broken, hopefully we will correct it soon.

I'm closing this as we already know it's broken.

**Comment by ernsyn** [comment]
@gmarziou 

git --version is 2.12.0

**Comment by ernsyn** [comment]
@jdubois I'm not sure whether I should create a new ticket or just comment on this one. 

I tried to upgrade from v4.0.7 to v4.0.8 using the fixed upgrade-generator but it's still failing with the same error as per the screenshots above.

Please let me know if you need me to open a new ticket.

**Comment by jdubois** [comment]
Thanks, for me this is still not working, I think there's a PR waiting on this, but we need time to code & test this again. This is why it's still marked everywhere as "not working", I know it's sad because I find this code really smart, but we just don't have time to do everything...

**Comment by ernsyn** [comment]
No worries @jdubois. Just thought I'd share what I found :)

I'll continue to upgrade manually for now.

**Comment by jdubois** [comment]
Yes thanks a lot @ernsyn - as I don't have the time on this, your feedback is very highly welcome. We need more help on this part...
